src/main/java/com/gooddata/md/MetadataService.java

```diff
@@ -94,6 +94,54 @@ public MetadataService(RestTemplate restTemplate) {
     }
 
     /**
+     * Remove metadata object URI
+     *
+     * @param obj metadata object to remove
+     * @param <T> type of the object to be returned
+     * @throws com.gooddata.md.ObjNotFoundException if metadata object not found
+     * @throws com.gooddata.GoodDataRestException   if GoodData REST API returns unexpected status code
+     * @throws com.gooddata.GoodDataException       if no response from API or client-side HTTP error
+     */
+    public <T extends Obj> void removeObj(T obj) {
```

Comment:
The type info is redundant `public void removeObj(Obj obj)` will do the same trick